cloud/ec2.go

```diff
@@ -622,22 +622,24 @@ func (m *ec2Manager) CheckInstanceType(ctx context.Context, instanceType string)
 
 // setNoExpiration changes whether a host should expire
 func (m *ec2Manager) setNoExpiration(ctx context.Context, h *host.Host, noExpiration bool) error {
-	resources, err := m.getResources(ctx, h)
-	if err != nil {
-		return errors.Wrap(err, "error getting host resources")
-	}
 	expireOnValue := expireInDays(evergreen.SpawnHostExpireDays)
-	_, err = m.client.CreateTags(ctx, &ec2.CreateTagsInput{
-		Resources: aws.StringSlice(resources),
-		Tags: []*ec2.Tag{
-			{
-				Key:   aws.String("expire-on"),
-				Value: aws.String(expireOnValue),
+	if !host.IsIntentHostId(h.Id) {
```

Comment:
Will this actually work? I'd think that you'd be able to set the host to never expire, but we won't ever tag it so the reaper will ☠

Reply:
We still set the expiration to the max spawn host value when the host is up: 
https://github.com/evergreen-ci/evergreen/blob/master/cloud/ec2_util.go#L157

By setting it to never expire, the job should keep extending it in the future.
https://github.com/evergreen-ci/evergreen/blob/master/model/host/host.go#L2232